cmd/kubeadm/app/cmd/phases/controlplanepreparejoin.go

```diff
@@ -0,0 +1,226 @@
+
+type controlPlanePrepareData interface {
+	Cfg() *kubeadmapi.JoinConfiguration
+	InitCfg() (*kubeadmapi.InitConfiguration, error)
+}
+
+func getControlPlanePreparePhaseFlags() []string {
+	return []string{
+		options.APIServerAdvertiseAddress,
+		options.APIServerBindPort,
+		options.CfgPath,
+		options.ControlPlane,
+		options.NodeName,
+		options.TokenDiscovery,
+		options.TokenDiscoveryCAHash,
+		options.TokenDiscoverySkipCAHash,
+	}
+}
+
+// NewControlPlanePreparePhase creates a kubeadm workflow phase that implements the preparation of the node to serve a control plane
+func NewControlPlanePreparePhase() workflow.Phase {
+	return workflow.Phase{
+		Name:  "control-plane-prepare",
+		Short: "Prepares the machine for serving a control plane.",
+		Long:  cmdutil.MacroCommandLongDescription,
+		Phases: []workflow.Phase{
+			newCertsSubphase(),
+			newKubeconfigSubphase(),
+			newManifestsSubphases(),
+			newCheckLocalEtcdStatusSubphase(),
+		},
+	}
+}
+
+func newCertsSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "certs",
+		Short:        "Generates the certificates for the new control plane components",
+		Run:          runCertsPhaseLocal,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newKubeconfigSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "kubeconfig",
+		Short:        "Generates the kubeconfig for the new control plane components",
+		Run:          runKubeconfigPhaseLocal,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newManifestsSubphases() workflow.Phase {
+	return workflow.Phase{
+		Name:  "manifests",
+		Short: "Generates the manifests for the new control plane components",
+		Phases: []workflow.Phase{
+			{
+				Name:           "all",
+				Short:          "Generates all static Pod manifest files",
+				InheritFlags:   getControlPlanePreparePhaseFlags(),
+				RunAllSiblings: true,
+			},
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeAPIServer),
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeControllerManager),
+			newControlPlanePrepareSubPhase(kubeadmconstants.KubeScheduler),
+		},
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newCheckLocalEtcdStatusSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:         "check-local-etcd-status",
+		Short:        "Checks the local etcd status",
+		Run:          runCheckLocalEtcdStatus,
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func newControlPlanePrepareSubPhase(component string) workflow.Phase {
+	return workflow.Phase{
+		Name:         controlPlanePhaseProperties[component].name,
+		Short:        controlPlanePhaseProperties[component].short,
+		Run:          runControlPlaneJoinSubPhase(component),
+		InheritFlags: getControlPlanePreparePhaseFlags(),
+	}
+}
+
+func runCertsPhaseLocal(c workflow.RunData) error {
+	data, ok := c.(controlPlanePrepareData)
+	if !ok {
+		return errors.New("certs phase invoked with an invalid data struct")
```

Comment:
our pattern in other files is to use the name of the parent phase in these messages - so `control-plane-prepare`.
e.g. https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/certs.go#L86
[1]

Reply:
I will adapt, thanks!

Reply:
🤔 
I just checked this and this is applied consistently to `fmt.Printf`, but not to errors.
Thinking about this, errors should be clean because the error could be concatenated other errors